### ISSUE TITLE
Multiple problems with insufficient privileges

### ISSUE THREAD

**Issue by rschoene** [tud-zih-energy/lo2s#133]
- [x] With `perf_event_paranoid` set to 3, we get this message instead of an helpful output: 
```
[29993381748090][pid: 23291][tid: 139717931222848][ERROR]: requested sampling event 'instructions' is not available!
```
- [x] print a more descriptive output when setting up the counter does not work. Here's what's said:
```
lo2s -e cpu/ref-cycles/ bash
[30014679462732][pid: 23301][tid: 139798486488896][ WARN]: cpu-clock isn't available, trying to use a fallback event
[30014680483668][pid: 23301][tid: 139798486488896][ERROR]: mapping memory for recording events failed. You can try to decrease the buffer size with the -m flag, or try to increase the amount of mappable memory by increasing /proc/sys/kernel/perf_event_mlock_kb
[30014688812406][pid: 23301][tid: 139798486488896][FATAL]: Aborting: Bad file descriptor
```
- [x] on `4.18.0-16-generic`, and default `perf_event_paranoid`, instructions, cycles, ... wont work, with `perf_event_paranoid==-1`, it works. Check why this is the case!

**Comment by rschoene** [comment]
The default for `perf_event_paranoid` seems to be `3`, which means disabling perf_events for users :frowning:

**Comment by tilsche** [comment]
This is specific to Ubuntu.

The mainline Linux only goes up to `2`.
Debian and Ubuntu both have patches such that paranoid goes up to `3` which disables `perf` completely and depending on some build option, the default is `3`. Meanwhile they failed to include the documentation update to reflect that.

https://lwn.net/Articles/696216/
https://www.kernel.org/doc/Documentation/sysctl/kernel.txt
https://salsa.debian.org/kernel-team/linux/blob/master/debian/patches/features/all/security-perf-allow-further-restriction-of-perf_event_open.patch
https://git.launchpad.net/~ubuntu-kernel/ubuntu/+source/linux/+git/bionic/commit/?id=d48d0c64e2895ad8eb6cabfbad586f8817fa1aab

**Comment by bmario** [comment]
> * print alternative events when default ("instructions") is/seems not available. Current output:
> 
> ```
> [29993381748090][pid: 23291][tid: 139717931222848][ERROR]: requested sampling event 'instructions' is not available!
> ```

To fix this message is not so easy and requires some refactoring.
There is already another place to catch problems in perf_event_open: 
https://github.com/tud-zih-energy/lo2s/blob/master/src/metric/perf_counter.cpp#L47

However, the message comes from here: 
https://github.com/tud-zih-energy/lo2s/blob/master/src/config.cpp#L450

Which is called in an earlier stage during initialization.

In the zebra world, we would refactor these two (and all the other) occurrences of `perf_event_open()`  to use one function, which will check and report on issues properly.

**Comment by bmario** [comment]
Fixed in ba9577fc4b22dbe4cac65c00691775cd3f507dbd and 1e5e6193cd83206c94327ad105441d422bf6d481
